oximachinerunner/__init__.py

```diff
@@ -265,7 +310,15 @@ def _run_oximachine(self, structure: Structure) -> OrderedDict:
         Returns:
             OrderedDict: with the keys metal_indices, metal_symbols,
                 prediction, max_probas, base_predictions
+
+        Raises:
+            NoMetalError: In case the structure does not contain a metal
         """
+        if not has_metal_sites(structure):
+            raise NoMetalError(
```

Comment:
One could consider making this a warning (and returning empty predictions) as nothing has gone "wrong" here - it's just that oximachine doesn't have anything to say about the structure.

Reply:
If so, it would be useful to have an `EMPTY_PREDICTION` constant (or similar) that is returned here, and that could also be imported and used by programs using the oximachinerunner to patch in an empty prediction when any of the other errors is encountered

Reply:
> If so, it would be useful to have an `EMPTY_PREDICTION` constant (or similar) that is returned here (i.e. just the usual dictionary with the right keys but with empty lists/tuples as values), and that could also be imported and used by programs using the oximachinerunner to patch in an empty prediction when any of the other errors is encountered

Reply:
i was not sure about what the "ideal" design for this would be---wherefore i did not merge it. 
I'm fine with returning an empty dictionary

Reply:
i didn't merge due to this point. was not sure what a good design would look like.

but since aiida-lsmo is the main user for now, i'll do what fits you ;) (and it seems also a fine choice to me)